pom.xml

```diff
@@ -695,6 +695,11 @@
       <artifactId>spotbugs-annotations</artifactId>
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-core</artifactId>
+      <scope>provided</scope>
+    </dependency>
```

Comment:
We don't need to use provided scope. That is intended for shipping packages that depend on stuff provided by their containerized environment, like a Maven plugin expected to run inside Maven, or a WAR intended to run inside an application server.

Instead, we need to explicitly depend on this as a `test` dependency in places where we need it, or as an optional runtime dependency. For us, the runtime dependency doesn't really matter because we only need it for the distribution tarball.

```suggestion
```

Reply:
This should be removed when #3000 is merged.